File: CS/CustomDocument/Custom DocumentManager/CustomDocument.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using DevExpress.Utils.Serializing;
using DevExpress.XtraBars.Docking2010.Views.Widget;

namespace CustomDocumentApplication {
    public class CustomDocument : Document {
        public CustomDocument() {

        }
        public CustomDocument(IContainer container)
            : base(container) {

        }
        public CustomDocument(IDocumentProperties defaultProperties)
            : base(defaultProperties) {

        }

        string customValue;
        [XtraSerializableProperty]
        public string CustomProperty {
            get { return customValue; }
            set {
                customValue = value;
            }
        }
    }
}

File: VB/CustomDocument/Custom DocumentManager/CustomDocument.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports DevExpress.Utils.Serializing
Imports DevExpress.XtraBars.Docking2010.Views.Widget

Namespace CustomDocumentApplication
	Public Class CustomDocument
		Inherits Document
		Public Sub New()

		End Sub
		Public Sub New(ByVal container As IContainer)
			MyBase.New(container)

		End Sub
		Public Sub New(ByVal defaultProperties As IDocumentProperties)
			MyBase.New(defaultProperties)

		End Sub

		Private customValue As String
		<XtraSerializableProperty> _
		Public Property CustomProperty() As String
			Get
				Return customValue
			End Get
			Set(ByVal value As String)
				customValue = value
			End Set
		End Property
	End Class
End Namespace
